bundles/org.openhab.binding.fmiweather/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,382 @@
+				<description>This is the weather forecast in 15 hours.</description>
+			</channel-group>
+			<channel-group id="forecastHours18" typeId="group-forecast">
+				<label>18 Hours Forecast</label>
+				<description>This is the weather forecast in 18 hours.</description>
+			</channel-group>
+			<channel-group id="forecastHours21" typeId="group-forecast">
+				<label>21 Hours Forecast</label>
+				<description>This is the weather forecast in 21 hours.</description>
+			</channel-group>
+			<channel-group id="forecastHours24" typeId="group-forecast">
+				<label>24 Hours Forecast</label>
+				<description>This is the weather forecast in 24 hours.</description>
+			</channel-group>
+			<channel-group id="forecastHours27" typeId="group-forecast">
+				<label>27 Hours Forecast</label>
+				<description>This is the weather forecast in 27 hours.</description>
+			</channel-group>
+			<channel-group id="forecastHours30" typeId="group-forecast">
+				<label>30 Hours Forecast</label>
+				<description>This is the weather forecast in 30 hours.</description>
+			</channel-group>
+			<channel-group id="forecastHours33" typeId="group-forecast">
+				<label>33 Hours Forecast</label>
+				<description>This is the weather forecast in 33 hours.</description>
+			</channel-group>
+			<channel-group id="forecastHours36" typeId="group-forecast">
+				<label>36 Hours Forecast</label>
+				<description>This is the weather forecast in 36 hours.</description>
+			</channel-group>
+			<channel-group id="forecastHours39" typeId="group-forecast">
+				<label>39 Hours Forecast</label>
+				<description>This is the weather forecast in 39 hours.</description>
+			</channel-group>
+			<channel-group id="forecastHours42" typeId="group-forecast">
+				<label>42 Hours Forecast</label>
+				<description>This is the weather forecast in 42 hours.</description>
+			</channel-group>
+			<channel-group id="forecastHours45" typeId="group-forecast">
+				<label>45 Hours Forecast</label>
+				<description>This is the weather forecast in 45 hours.</description>
+			</channel-group>
+			<channel-group id="forecastHours48" typeId="group-forecast">
+				<label>48 Hours Forecast</label>
+				<description>This is the weather forecast in 48 hours.</description>
+			</channel-group>
+			<channel-group id="forecastHours51" typeId="group-forecast">
+				<label>51 Hours Forecast</label>
+				<description>This is the weather forecast in 51 hours.</description>
+			</channel-group>
+			<channel-group id="forecastHours54" typeId="group-forecast">
+				<label>54 Hours Forecast</label>
+				<description>This is the weather forecast in 54 hours.</description>
+			</channel-group>
+		</channel-groups>
+
+		<config-description>
+			<parameter name="location" type="text" required="true">
+				<label>Location</label>
+				<description>Location of weather in geographical coordinates (latitude,longitude).</description>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+
+
+	<!-- Groups -->
```

Comment:
How did you create the selection of channels? As I looked into the referenced [webpage(s)](http://opendata.fmi.fi/meta?observableProperty=forecast) there are many more channels.

Reply:
It's a good point, I think I picked the most "common" / "relevant" channels but perhaps I could pick all.

In the document they have an example query similar to below, not restricting any parameters in the call:
http://opendata.fmi.fi/wfs?service=WFS&version=2.0.0&request=getFeature&storedquery_id=fmi::forecast::hirlam::surface::point::multipointcoverage&place=jaala&latlon=60.1,19.9&

Resulting in these parameters (bold ones included in the binding):

1. GeopHeight [not related to forecasts really]
2. **Temperature**
3. **Pressure**
4. **Humidity**
5. **WindDirection**
6. **WindSpeedMS**
7. WindUMS  [redundant if one has direction and m/s speed]
8. WindVMS  [redundant if one has direction and m/s speed]
9. MaximumWind [maximum taken from 6 short-term-speed-averages-of-10min?]
10. **WindGust**
11. DewPoint [would be useful?]
12. **TotalCloudCover**
13. **WeatherSymbol3**
14. LowCloudCover [does user care about this detailed cloud information when TotalCloudCover is available?]
15. MediumCloudCover [does user care about this detailed cloud information when TotalCloudCover is available?]
16. HighCloudCover [does user care about this detailed cloud information when TotalCloudCover is available?]
17. **Precipitation1h**
18. PrecipitationAmount [what is the meaning of this?]
19. RadiationGlobalAccumulation
20. RadiationLWAccumulation
21. RadiationNetSurfaceLWAccumulation
22. RadiationNetSurfaceSWAccumulation
23. RadiationDiffuseAccumulation
24. LandSeaMask [not related to forecasts really]

The [API](http://opendata.fmi.fi/meta?observableProperty=forecast) you linked gives even more parameters such as 
```xml
<component>
  <ObservableProperty gml:id="cloud_indicator">
      <label>Lightning type indicator</label>
      <basePhenomenon>Lightning</basePhenomenon>
      <uom uom="Index"/>
  </ObservableProperty>
</component>
```

But it is not queriable at least with `storedquery_id=fmi::forecast::hirlam::surface::point::multipointcoverage`:
http://opendata.fmi.fi/wfs?service=WFS&version=2.0.0&request=getFeature&storedquery_id=fmi::forecast::hirlam::surface::point::multipointcoverage&place=jaala&latlon=60.1,19.9&parameters=cloud_indicator

```xml
<Exception exceptionCode="OperationParsingFailed">

   <ExceptionText>Unknown parameter &apos;cloud_indicator&apos;!</ExceptionText>
   <ExceptionText>URI: /wfs?latlon=60.1%2C19.9&amp;parameters=cloud_indicator&amp;place=jaala&amp;request=getFeature&amp;service=WFS&amp;storedquery_id=fmi%3A%3Aforecast%3A%3Ahirlam%3A%3Asurface%3A%3Apoint%3A%3Amultipointcoverage&amp;version=2.0.0</ExceptionText>
    
</Exception>
```

So I'm not sure what is the full set of available parameters but certainly we could add more.

Do you see anything relevant?


P.S. as side note I have been pondering of adding additional FMI data sources such as air quality, road condition warnings etc. but perhaps that is for a later date.

Reply:
Honestly I was coming across that as I was (incorrectly) providing some non-'weather' stations and was asking myself whether they could provide some usefull observations for Finnish users.
Regarding the forecast I would agree that it felt to me like you selected the most important ones but was not sure whether there were probably forecasts on some topics that are not provided by other weather bindings.